File: backend/product-service/src/main/java/com/mall/product/entity/Banner.java
package com.mall.product.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("banner")
public class Banner {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String subtitle;
    private String description;
    private String imageUrl;
    private String linkUrl;
    @TableField("btn_text")
    private String btnText;
    private String align;
    private Integer sort;
    private Integer status;
    private LocalDateTime createdTime;
}

File: frontend/src/admin/api/banner.ts
import request from './request'

export interface Banner {
  id: number
  title: string
  subtitle: string | null
  description: string | null
  imageUrl: string
  linkUrl: string
  btnText: string | null
  align: string
  sort: number
  status: number
  createdTime: string
}

export function getBannerList() {
  return request.get<any, Banner[]>('/product/banner/admin/list')
}

export function addBanner(data: Partial<Banner>) {
  return request.post<void>('/product/banner/admin/add', data)
}

export function updateBanner(data: Partial<Banner>) {
  return request.put<void>('/product/banner/admin/update', data)
}

export function deleteBanner(id: number) {
  return request.delete<void>('/product/banner/admin/delete/' + id)
}
